### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by JJSanch**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -16,23 +21,41 @@
...
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is a president of the galaxy.", "Arthus Dent is a radio employee."]
 
 
 
 
 
 // b) Create the function that makes the test pass.
+//attempted to use higher order function to map across the array and join the sentences together as string pushed into one array to get expected output
 
+const capNameArray = (array) =>  array.map(arrayItem => typeof arrayItem === "string").push.join("")
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:55]: I can see where you're going with this but we have a couple of data types we're dealing with mainly: strings as values inside of objects in an array. Whats the pseudo code logic that you want to use here to solve this problem? Comment below--

File: code-challenges.test.js [hunk 2]
```diff
@@ -44,14 +67,27 @@ var hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
...
+    })
+})
+    
 
 // b) Create the function that makes the test pass.
 
-
-
+const remaindersOnly = (array) =>  array.map(arrayItem => arrayItem % 3)
+//created a function that maps across arrays and returns the remainder of numbers divided by 3.  Was not able to get the test to pass
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:90]: This is pretty close! when I run the code I get this 
[2, NaN, 0, -1, 0, NaN, 0]
[2, NaN, 1, -1, NaN, 1]
So really close! 
What tool can we use to get rid of the things that become Not-A-Number?

File: code-challenges.test.js [hunk 3]
```diff
@@ -61,10 +97,28 @@ var cubeAndSum1 = [2, 3, 4]
...
+      expect(numbersCubed(cubeAndSum1)).toEqual(expectedNumber1)
+      expect(numbersCubed(cubeAndSum2)).toEqual(expectedNumber2)
+    })
+})
 
 
 
 // b) Create the function that makes the test pass.
+//created a function that takes in the input arrays and returns the sum of all numbers cubed.  However was not able to get test to pass.
+const numbersCubed = (array) =>  array.map(arrayItem => typeof arrayItem * 3).sum
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:124]: (Pseudo Code) What could you change here to make this function work?

File: interview-questions.md [hunk 4]
```diff
@@ -7,44 +7,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
+               db:migrate - this migrates changes you made to the a rails databse
+
+  Researched answer: 
+                     rails db:create - sets up an empty databse
+
+                     rails new app_name - generates new folder with enture Rails 
+                     directory sructure.  All code needed to run app is included in this command
+
+                     rails generate rseource - sets up a migration and a model needed to use something as an ActiveRecord class
```
Comment (WalkerTexas-coder) [inline interview-questions.md:40]: this last line here is what the question is talking about
What were the other generate commands?

File: interview-questions.md [hunk 5]
```diff
@@ -7,44 +7,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
+  6. As a user, I can see a form to create a new to do list.
+
+  7. As a user, I can naivgate out of the form and back to the home page.
+
+  8. As a user, I can click on a button that will save my created to do list.
+
+  9. As a user, I can click on a button that will edit an existing to do list.
+
+  10. As a user, I can click on the title of the app at the top of the screen and be routed to the app home page.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:96]: EGG-CELL-ENT!!!

File: rails-commenting.rb [hunk 6]
```diff
@@ -55,14 +62,17 @@ def destroy
       redirect_to blog_posts_path
     else
       # ---8)
+      # This will allow for deletion of a specific blog post
       redirect_to blog_post_path(@post)
     end
   end
 
   # ---9)
+  #Private is used to list the strong params.  Restricts the scope of where a method can be called. Everything below the private keyword is included in its protection, so this is typically the last method inside the controller
```
Comment (WalkerTexas-coder) [inline rails-commenting.rb:71]: Great Stuff!

File: rails-commenting.rb [hunk 7]
```diff
@@ -34,13 +39,15 @@ def create
   end
 
   # ---6)
+  #This will allow a user to edit an already created blogpost
```
Comment (WalkerTexas-coder) [inline rails-commenting.rb:42]: Perfect!

File: rails-commenting.rb [hunk 8]
```diff
@@ -7,24 +7,29 @@
...
 class BlogPostsController < ApplicationController
   def index
     # ---2)
+    # this method will access all items from the database.  This is an Active Record call that will save all the content from the db as an instance variable
     @posts = BlogPost.all
   end
 
   def show
     # ---3)
+    # this looks for one item in the databse.  It is done by accessing the ID of the item
```
Comment (WalkerTexas-coder) [inline rails-commenting.rb:20]: Yes!

**Review by WalkerTexas-coder — COMMENTED** [review]
Okay JJ, I left some comments for you to "reply" to. No need to update your code just reply in the comment sections. I liked your logic, you have some really solid ideas about how to accomplish these challenges. Your rails answers are great and your interview answers are pretty good to my man!

File: code-challenges.test.js (hunk 1 above)
Comment (JJSanch) [reply code-challenges.test.js:55]: //map across an array using the .map higher order function
//identify the mapped item as a string using the typeof operator
//push the mapped array into a new array
//use .join to join the returned values into a string

File: code-challenges.test.js (hunk 2 above)
Comment (JJSanch) [reply code-challenges.test.js:90]: .filter() method

File: code-challenges.test.js (hunk 3 above)
Comment (JJSanch) [reply code-challenges.test.js:124]: value ** 3

File: interview-questions.md (hunk 4 above)
Comment (JJSanch) [reply interview-questions.md:40]: g Model
g Resource
g Controller

**Pull request merged**